### PR TITLE
[hotfix-1.43] Upgrade http client library 'got' to latest version

### PR DESCRIPTION
**Pull request by holgerkoser**
**What this PR does / why we need it**:
We have observed situations where the `kube-apiserver` is throttling requests from the dashboard. In the log we see that a request does not return and the reflector loop is blocked.
log
There are some closed iusses in `got` (e.g. [1341](https://github.com/sindresorhus/got/issues/1341), [1295](https://github.com/sindresorhus/got/issues/1295) and [1287](https://github.com/sindresorhus/got/issues/1287)) which describe problems with hanging requests. These issues could be the reason for cache expiration problems in the dashboard backend.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Upgrade http client library `got` to latest version to get rid of cache expiration problems.
```

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-1.43.3-dev
\ No newline at end of file
+1.43.2
```
Comment (petersutter) [inline VERSION:1]: revert change

Comment (holgerkoser) [reply VERSION:1]: Ja.Habich auch gerade gesehen. 👍

File: backend/lib/kubernetes-client/cache/Reflector.js [hunk 2]
```diff
@@ -133,29 +121,52 @@ class Reflector {
...
+      logger.debug('Closing socket of %s has been triggered', this.expectedTypeName)
+      await pEvent(this.socket, 'close', { timeout })
+      logger.debug('Closed socket of %s', this.expectedTypeName)
+    } catch (err) {
+      logger.debug('Terminating socket of %s', this.expectedTypeName)
+      this.socket.terminate()
+      logger.debug('Terminated socket of %s', this.expectedTypeName)
+    }
+  }
+
```
Comment (petersutter) [inline backend/lib/kubernetes-client/cache/Reflector.js:142]: I guess we don't need the excessive log here in this function anymore, right?
Also you could move `await pEvent..` on top of `this.socket.close`, just to be sure (as you had it previously)

File: backend/lib/kubernetes-client/cache/Reflector.js [hunk 3]
```diff
@@ -272,7 +288,9 @@ class Reflector {
           return
         }
       } finally {
-        await closeOrTerminateSocket(this.socket, this.gracePeriod.asMilliseconds())
+        logger.info('Trying to gracefully close socket of %s', this.expectedTypeName)
+        await this.closeOrTerminateSocket()
+        logger.info('Closed or terminated socket of %s', this.expectedTypeName)
```
Comment (petersutter) [inline backend/lib/kubernetes-client/cache/Reflector.js:293]: remove logs or at least one of them

File: backend/lib/kubernetes-client/cache/Reflector.js (hunk 2 above)
Comment (holgerkoser) [reply backend/lib/kubernetes-client/cache/Reflector.js]: ✔️

File: backend/lib/kubernetes-client/cache/Reflector.js (hunk 3 above)
Comment (holgerkoser) [reply backend/lib/kubernetes-client/cache/Reflector.js]: ✔️

**Pull request merged**